framework/test-vectors/key-description.md

```diff
@@ -0,0 +1,147 @@
+[//]: # "Copyright Amazon.com Inc. or its affiliates. All Rights Reserved."
+[//]: # "SPDX-License-Identifier: CC-BY-SA-4.0"
+
+# Key Description
+
+## Version
+
+1.0.0
+
+## Dependencies
+
+This serves as a reference of all features that this feature depends on.
+
+| Feature                             | Min Version | Max Version |
+| ----------------------------------- | ----------- | ----------- |
+| [keys-manifest](./keys-manifest.md) | 1           | n/a         |
+
+## Summary
+
+The Key Description structure defines a standard way of describing
+keyring configuration as part of a manifest.
+The AwsCryptographicMaterialProvidersLibrary describes a common way
+to configured key distribution.
+These key descriptions explain how to instantiate the required [keyring](../keyring-interface.md)
+or [cmm](../cmm-interface.md).
+
+## Out of Scope
+
+This file does not define any actual manifests,
+only a common structure that is used in Crypto Tools libraries manifest definitions
+that need to include a description of a key description.
+
+## Motivation
+
+Several manifests have a requirement to define keyrings that need to
+be constructed while processing that manifest.
+In order to maintain consistency for these definitions across manifest types,
+this file defines how keyrings are defined in any
+manifest type that needs to describe keyring configuration.
+
+There are also legacy test vectors that exist.
+This key description MUST be compatible
+with these legacy objects.
+These objects were referred to as Master Keys
+and Master Key Providers.
+
+## Guide-level Explanation
+
+The master key structure is a JSON structure that identifies necessary characteristics
+about an AwsCryptographicMaterialProvidersLibrary keyring.
+
+## Reference-level Explanation
+
+### Contents
+
+A key description structure is defined as a JSON object with the following members:
+
+- `type` : Type of keyring/cmm
```

Comment:
nit: although when using a keyring with no cmm it will default to a `default-cmm` should we add `default-cmm` as an allowed value? what if someone implements their own key description json file and they add `default-cmm` as the type? should it get thrown out?